crud/src/main/java/com/apirest/crud/Service/userImplService.java

```diff
@@ -16,24 +16,56 @@ public class userImplService implements userService {
     private userRepository userrepository;
 
     @Override
-    public user save(user u) {
-
-        return userrepository.save(u);
+    public User save(User u) {
+        User obj = userrepository.save(u);
+        if (obj == null) {
+            throw new NullPointerException("user not found");
+        }
+        return obj;
     }
+    
+    
     @Override
-    public user getUserById(Long id) {
-       Optional<user> optionalUser= userrepository.findById(id);
-       return optionalUser.get();
+    public Optional<User> getUserById(Long id) {
+       Optional<User> obj= userrepository.findById(id);
+        if (obj == null) {
+             throw new NullPointerException("user not exists");
+        }
+        return obj;
     }
 
     @Override
-    public List<user> getAll() {
-     return userrepository.findAll();
+    public List<User> getAll() {
+        List obj=userrepository.findAll();
+        if (obj == null) {
+             throw new NullPointerException("users not exists");
+        }
+        return  obj;
     }
 
     @Override
-    public void deleteUser(Long id) {
-         userrepository.deleteById(id);
+    public boolean deleteUser(Long id) {
+        try {
+            if(userrepository.existsById(id)){
+                userrepository.deleteById(id);
+                return true;
+            }
+            return false;
+        } catch (Exception e) {
+            throw new NullPointerException("the object witch ID " + id + " not exists");
+        }
+
+
     }
-    
+
+    @Override
+    public User updateUser(Long id,User user) {
+
+            if(!userrepository.existsById(id)) {
+              throw new NullPointerException  ("the user not exists");

```

Comment:
please use another exception, this is not the proper one, probably user not found excp